core/lib/Thelia/Form/HookCreationForm.php

```diff
@@ -81,6 +87,19 @@ protected function buildForm()
         ;
     }
 
+    public function checkCodeUnicity($code, ExecutionContextInterface $context){
+        $count = HookQuery::create()->filterByCode($code)->count();
```

Comment:
You have to check the type of the hook too, as you can have two hooks with the same code, one used in the front office, the other in the back office.